### ISSUE TITLE
Facter command-line errors

### ISSUE THREAD

**Issue by m8051** [puppetlabs/facter-ng#106]
## Describe the Bug

Hi @IrimieBogdan  I am not sure if the bug template is the right way to report the issue I am having so please feel free to make any chances you might consider.

I've recently switched from Ubuntu 18 to Debian 10.1. I meant to keep working on a few issues you guys have but I am facing a few errors while executing facter-ng with any fact on Debian.

Thanks in advance.

## Expected Behavior

```bash
jordi@dark-nebula 11:40:24 ~/repos/facter-ng |master ✓| →  ./bin/facter-ng timezone
IST
```

## Steps to Reproduce

1 - Cloning the repository:

```bash
jordi@dark-nebula 11:26:43 ~/repos →  git clone git@github.com:m8051/facter-ng.git                                                                                       
Cloning into 'facter-ng'...                                                                                                                                              
remote: Enumerating objects: 4397, done.                                                                                                                                 
remote: Total 4397 (delta 0), reused 0 (delta 0), pack-reused 4397                                                                                                       
Receiving objects: 100% (4397/4397), 584.22 KiB | 1.51 MiB/s, done.                                                                                                      
Resolving deltas: 100% (2930/2930), done.             
```
2 - Installing and managing facter-ng gems:

```bash
jordi@dark-nebula 11:27:34 ~/repos/facter-ng |master ✓| →  bundle install
Fetching gem metadata from https://rubygems.org/........
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Fetching rake 10.5.0
Installing rake 10.5.0
Fetching public_suffix 4.0.1
Installing public_suffix 4.0.1
Fetching addressable 2.7.0
Installing addressable 2.7.0
Fetching ast 2.4.0
Installing ast 2.4.0
Fetching thread_safe 0.3.6
Installing thread_safe 0.3.6
Fetching descendants_tracker 0.0.4
Installing descendants_tracker 0.0.4
Fetching ice_nine 0.11.2
Installing ice_nine 0.11.2
Fetching axiom-types 0.1.1
Installing axiom-types 0.1.1
Using bundler 2.0.2
Fetching coercible 1.0.0
Installing coercible 1.0.0
Fetching equalizer 0.0.11
Installing equalizer 0.0.11
Fetching virtus 1.0.5
Installing virtus 1.0.5
Fetching codeclimate-engine-rb 0.4.1
Installing codeclimate-engine-rb 0.4.1
Fetching json 2.2.0
Installing json 2.2.0 with native extensions
Fetching docile 1.3.2
Installing docile 1.3.2
Fetching simplecov-html 0.10.2
Installing simplecov-html 0.10.2
Fetching simplecov 0.16.1
Installing simplecov 0.16.1
Fetching tins 1.21.1
Installing tins 1.21.1
Fetching term-ansicolor 1.7.1
Installing term-ansicolor 1.7.1
Fetching thor 0.20.3
Installing thor 0.20.3
Fetching coveralls 0.8.23
Installing coveralls 0.8.23
Fetching diff-lcs 1.3
Installing diff-lcs 1.3
Fetching erubis 2.7.0
Installing erubis 2.7.0
Using facter-ng 0.0.10 from source at `.`
Fetching path_expander 1.1.0
Installing path_expander 1.1.0
Fetching sexp_processor 4.13.0
Installing sexp_processor 4.13.0
Fetching ruby_parser 3.14.0
Installing ruby_parser 3.14.0
Fetching flay 2.12.1
Installing flay 2.12.1
Fetching flog 4.6.3
Installing flog 4.6.3
Fetching jaro_winkler 1.5.3
Installing jaro_winkler 1.5.3 with native extensions
Fetching kwalify 0.7.2
Installing kwalify 0.7.2
Fetching launchy 2.4.3
Installing launchy 2.4.3
Fetching parallel 1.18.0
Installing parallel 1.18.0
Fetching parser 2.6.5.0
Installing parser 2.6.5.0
Fetching psych 3.1.0
Installing psych 3.1.0 with native extensions
Fetching rainbow 3.0.0
Installing rainbow 3.0.0
Fetching reek 5.4.0
Installing reek 5.4.0
Fetching rspec-support 3.9.0
Installing rspec-support 3.9.0
Fetching rspec-core 3.9.0
Installing rspec-core 3.9.0
Fetching rspec-expectations 3.9.0
Installing rspec-expectations 3.9.0
Fetching rspec-mocks 3.9.0
Installing rspec-mocks 3.9.0
Fetching rspec 3.9.0
Installing rspec 3.9.0
Fetching ruby-progressbar 1.10.1
Installing ruby-progressbar 1.10.1
Fetching unicode-display_width 1.6.0
Installing unicode-display_width 1.6.0
Fetching rubocop 0.74.0
Installing rubocop 0.74.0
Fetching tty-which 0.4.1
Installing tty-which 0.4.1
Fetching rubycritic 4.1.0
Installing rubycritic 4.1.0
Bundle complete! 7 Gemfile dependencies, 47 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
```

3 - Executing facter

```bash
jordi@dark-nebula 11:43:06 ~/repos/facter-ng |master ✓| →  ./bin/facter-ng
No LSB modules are available.
my_custom_fact => "my_custom_os",
my_external_fact => "123",
os => "my_custom_os"
```

**This caught my attention as facter-ng barely shows any information**

4 - Executing facter-ng with timezone fact

```bash
jordi@dark-nebula 11:43:31 ~/repos/facter-ng |master ✓| →  ./bin/facter-ng timezone
Traceback (most recent call last):
        10: from ./bin/facter-ng:6:in `<main>'
         9: from /home/jordi/.rvm/rubies/ruby-2.5.1/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:54:in `require'
         8: from /home/jordi/.rvm/rubies/ruby-2.5.1/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:54:in `require'
         7: from /home/jordi/repos/facter-ng/lib/framework/cli/cli_launcher.rb:11:in `<top (required)>'
         6: from /home/jordi/.rvm/gems/ruby-2.5.1/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
         5: from /home/jordi/.rvm/gems/ruby-2.5.1/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
         4: from /home/jordi/.rvm/gems/ruby-2.5.1/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
         3: from /home/jordi/.rvm/gems/ruby-2.5.1/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
         2: from /home/jordi/repos/facter-ng/lib/framework/cli/cli.rb:101:in `query'
         1: from /home/jordi/repos/facter-ng/lib/facter-ng.rb:19:in `to_user_output'
/home/jordi/repos/facter-ng/lib/framework/formatters/hocon_fact_formatter.rb:15:in `format': undefined method `empty?' for nil:NilClass (NoMethodError)
```
## Environment

 - Ruby Version: ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux]
 - Platform Debian Buster 10.1

## Additional Context
TBD

**Comment by oanatmaria** [comment]
Hi,

Thank you for this feedback. This project is still work in progress, so on Debian there are no many facts implemented. https://github.com/puppetlabs/facter-ng/tree/master/lib/facts/debian 

If you would like, you can implement timezone fact on debian similar with the one existing on ubuntu.

We will investigate the error tho because it isn't normal for the app to crash when a fact is non existent.

**Comment by IrimieBogdan** [comment]
@m8051 Initially we wanted to have Debian and Ubuntu as distinct OS-es, but they are very similar and we should give it a try and treat them both as one OS. In order to do this @sebastian-miclea made a PR #107  that will consider Debian as Ubuntu. If this is a success, we will merge Debian and Ubuntu. 

As next steps:
- we will merge #107 this in master, so please give master a try on Debian
- for the moment please continue to generate facts for Ubuntu, even if you are using Debian, and if the experiment is a success, we will merge Debian and Ubuntu.

I will leave this opened in order to investigate the exception you encountered. Even if the facts does not exist, you should have received an empty string.

**Comment by m8051** [comment]
Thanks @oanatmaria and @IrimieBogdan for looking after it.
I'd like to continue contributing and I am thinking of having a dockerized dev environment running Ubuntu so I can test new implementations ... not sure tho if that would be of any useful for you guys ...

**Comment by IrimieBogdan** [comment]
@m8051 it's great that you are supporting this project! #107 was merged. Regarding docker, if you can share the dev environment I can have a look and give it a try.

**Comment by IrimieBogdan** [comment]
@m8051  in this [table](https://docs.google.com/spreadsheets/d/1nLFU9_TEcSismnFqCySLBHdWfE-hcojuO54LWblGlGU/edit?usp=sharing) we keep track of facts for multiple platforms.

**Comment by m8051** [comment]
> @m8051 it's great that you are supporting this project! #107 was merged. Regarding docker, if you can share the dev environment I can have a look and give it a try.

**Comment by m8051** [comment]
> @m8051 it's great that you are supporting this project! #107 was merged. Regarding docker, if you can share the dev environment I can have a look and give it a try.

Thanks @IrimieBogdan, the #107 has fixed indeed the issues I had on Debian.

```bash
jordi@dark-nebula 05:13:51 ~/repos/facter-ng |master ✓| →  ./bin/facter-ng timezone
IST
```
```bash
jordi@dark-nebula 05:16:36 ~/repos/facter-ng |master ✓| →  ./bin/facter-ng os.distro
No LSB modules are available.
{
  codename => "buster",
  description => "Debian GNU/Linux 10 (buster)",
  id => "Debian",
  release => {
    full => "10",
    major => "10",
    minor => null
  }
}
```
Regarding sharing my dev environment, I started to work on a Dockerfile to containerize the app on but I stopped as I realized it might not be a good approach as Docker runs under the host system kernel and the outcome will be most likely the same.

You guys use Docker or any other platform (Vagrant) to create and configure reproducible environments?

**Comment by m8051** [comment]
> @m8051 in this [table](https://docs.google.com/spreadsheets/d/1nLFU9_TEcSismnFqCySLBHdWfE-hcojuO54LWblGlGU/edit?usp=sharing) we keep track of facts for multiple platforms.

Awesome!! Thanks for sharing, helps to have a better insight and understanding of it.

**Comment by IrimieBogdan** [comment]
@m8051 in order to develop/test Facter we use VMs or bare mettle (where virtualisation is not possible). The VM's are provisioned by [vmpooler](https://github.com/puppetlabs/vmpooler). As Facter usually runs on a VM or bare mettle in production, I would encourage the use of VMs wherever possible for development (as you mentioned, Vagrant could be a nice option).

**Comment by IrimieBogdan** [comment]
The error that was reported in this issue should be resolved by #113
